src/verilog.y

```diff
@@ -2628,22 +2657,59 @@ assignOne<nodep>:
                 variable_lvalue '=' expr                { $$ = new AstAssignW($2,$1,$3); }
         ;
 
-//UNSUPdelay_or_event_controlE<nodep>:  // IEEE: delay_or_event_control plus empty
-//UNSUP         /* empty */                             { $$ = nullptr; }
-//UNSUP |       delay_control                           { $$ = $1; }
-//UNSUP |       event_control                           { $$ = $1; }
-//UNSUP |       yREPEAT '(' expr ')' event_control      { }
-//UNSUP ;
+delay_or_event_controlE<nodep>:  // IEEE: delay_or_event_control plus empty
+                /* empty */				{ $$ = nullptr; }
+        |        delay_control
+                        {
+                            if ($1->fileline()->timingOn()) {
+                                 if (v3Global.opt.timing()) {
+                                     $$ = $1;
+                                 } else {
+                                     DLYNOTIMING($1);
+                                     $$ = nullptr;
+                                 }
+                             } else {
+                                 DEL($1);
+                                 $$ = nullptr;
+                             }
+            }
```

Comment:
In verilog.y we generally scrunch the number of code lines so it's easier to still read the grammar.  Alternatively make a function and call it.